frontend/src/pages/adminRoles.js

```diff
@@ -1,64 +1,119 @@
-              </td>
-              {this.state.isEditing && (
-                <td>
-                  <Button>Submit</Button>
-                </td>
-              )}
-            </tr>
-          </tbody>
-        </Table>
-        <Button
-          variant="primary"
-          disabled={this.state.isEditing}
-          onClick={() => this.setState({ isEditing: true })}
-        >
-          Edit
-        </Button>
-        <Button
-          variant="info"
-          disabled={!this.state.isEditing}
-          onClick={() => this.setState({ isEditing: false })}
-        >
-          Save
-        </Button>
+        <Card className="admin-roles-card">
+          <CardBody>
+            <Table size="sm" hover className="candidate-table">
+              <thead class="thead-dark">
+                <tr>
+                  <th>First Name</th>
+                  <th>Last Name</th>
+                  <th>Email</th>
+                  <th>Role</th>
+                  {isEditing && <th>Submit</th>}
+                </tr>
+              </thead>
+              <tbody>
+                {users.map((user, i) => {
+                  return (
+                    <>
+                      <tr key={i}>
+                        <td>{user.firstName}</td>
+                        <td>{user.lastName}</td>
+                        <td>{user.email}</td>
+                        <td>
+                          {isEditing ? (
+                            <ChangeRole
+                              value={user.role}
+                              handleChange={e => {
+                                this.handleRoleChange(e, i)
+                              }}
+                            />
+                          ) : (
+                            user.role
+                          )}
+                        </td>
+                        {isEditing && (
+                          <td>
+                            <Button
+                              color="success"
+                              size="sm"
+                              onClick={() => this.handleRoleSubmit(user.email)}
+                              disabled={!(newRole != -1 && selectedUser === i)}
+                            >
+                              Submit
+                            </Button>
+                          </td>
+                        )}
+                      </tr>
+                    </>
+                  )
+                })}
+              </tbody>
+            </Table>
+            <Button
+              variant="primary"
+              disabled={isEditing}
+              onClick={() => {
+                this.setState({ isEditing: true })
+              }}
+            >
+              Edit
+            </Button>
+            <Button
+              variant="info"
+              disabled={!isEditing}
+              onClick={() => this.setState({ isEditing: false })}
+            >
+              Save
+            </Button>
```

Comment:
Also not a huge priority, but each entry of the table has a `Submit` button, but the table as a whole also has a `Save` button.  The submit button actually completes the change, but the `Save` button simply changes the mode back to viewing. An easy fix would be renaming this to `View` or something. Eventually maybe merging the edit and view functionality, but its fine to keep it for now.